patches/minecraft/net/minecraft/client/multiplayer/ClientChunkProvider.java.patch

```diff
@@ -1,18 +1,10 @@
 --- a/net/minecraft/client/multiplayer/ClientChunkProvider.java
 +++ b/net/minecraft/client/multiplayer/ClientChunkProvider.java
-@@ -61,6 +61,7 @@
-          int i = this.field_217256_d.func_217191_a(p_73234_1_, p_73234_2_);
-          Chunk chunk = this.field_217256_d.func_217192_a(i);
-          if (func_217249_a(chunk, p_73234_1_, p_73234_2_)) {
-+            net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.world.ChunkEvent.Unload(chunk));
-             this.field_217256_d.func_217190_a(i, chunk, (Chunk)null);
```

Comment:
Moving this to where you did is wrong, as this event should be fired BEFORE the chunk is removed from the world.